deepmd/infer/deep_pot.py

```diff
@@ -498,7 +519,10 @@ def _eval_inner(
             av = np.reshape(av, [nframes, natoms, 9])
             return energy, force, virial, ae, av
         else:
-            return energy, force, virial
+            if self.ntypes_spin == 0:
+                return energy, force, virial
+            else:
+                return energy, force, virial, natoms_vec
```

Comment:
This is a breaking change of the `DeepPot.eval`. I would not recommend doing so.

Please add a new method to get the `natoms_vec`.

Reply:
We solved this problem.